### ISSUE TITLE
Issues with attaching file in 9.1

### ISSUE THREAD

**Issue by gloomy-master** [glpi-project/glpi#1101]
Issues with attaching files.
1.  When file is attached automatically from email  GLPI randomly chooses any technician as a person who attached it. How to fix it?
2.  We receive  files which have a name in Russian. For a example: УИЗО.docx, with this name it cannot be attached as unknown format. If I add a digit, for example УИЗО1.docx, it is added like this  - 1.docx. How to fix it?

**Comment by orthagh** [comment]
Hello @gloomy-master 

1. The user is associated to the document and not on the relation with the ticket.
We can have some scenario with mailgate:
- The document is a new one (never uploaded on glpi), it will be associated to a virtual user (requester) in ticket timeline
- The document already exists and associated to the previous user. I think this is your case. You can check in the document form, the historical tab.

2. I can't replicate with a test file named like your example. Could you attach an example file (without personal data) ?

**Comment by gloomy-master** [comment]
> 
The documents can have the same names as we use different word or excel templates, which user fills and sends it as an email with attachment to glpi. I think it would be enough if all docs which are added automatically would have associated user, for example, the System.
> 
The documents can have the same names as we use different word or excel templates, which user fills and sends it as an email with attachment to glpi. I think it would be enough if all docs which are added automatically would have associated user, for example, the System.

> 
I can't replicate with a test file named like your example. Could you attach an example file (without personal data) ?
> 
Please, check the attachment.

**Comment by tomolimo** [comment]
to attach the file, zip it!, and it should be accepted by github, if not just rename it `yourname.yourextension.txt`

**Comment by gloomy-master** [comment]
Done
[УИЗО.zip](https://github.com/glpi-project/glpi/files/515435/default.zip)

**Comment by gloomy-master** [comment]
Have any news on this issue?

**Comment by AnaelMobilia** [comment]
I just tried : I can attache the document (the send is say as complete), but when I try to save, I got an error (linked to #1511 ?)
```
2017-01-03 19:42:28 [2@TOUR-ANAEL]
  *** PHP Notice(8): Undefined index: linkfield
  Backtrace :
  inc\log.class.php:151                              
  inc\commondbtm.class.php:381                       Log::constructHistory()
  inc\commondbtm.class.php:1146                      CommonDBTM->updateInDB()
  inc\document_item.class.php:163                    CommonDBTM->update()
  inc\commondbtm.class.php:853                       Document_Item->post_addItem()
  inc\commonitilobject.class.php:1745                CommonDBTM->add()
  inc\commonitilobject.class.php:585                 CommonITILObject->addFiles()
  inc\ticket.class.php:1079                          CommonITILObject->prepareInputForUpdate()
  inc\commondbtm.class.php:1065                      Ticket->prepareInputForUpdate()
  front\ticket.form.php:60                           CommonDBTM->update()
2017-01-03 19:42:28 [2@TOUR-ANAEL]
  *** PHP Notice(8): Undefined index: field
  Backtrace :
  inc\log.class.php:152                              
  inc\commondbtm.class.php:381                       Log::constructHistory()
  inc\commondbtm.class.php:1146                      CommonDBTM->updateInDB()
  inc\document_item.class.php:163                    CommonDBTM->update()
  inc\commondbtm.class.php:853                       Document_Item->post_addItem()
  inc\commonitilobject.class.php:1745                CommonDBTM->add()
  inc\commonitilobject.class.php:585                 CommonITILObject->addFiles()
  inc\ticket.class.php:1079                          CommonITILObject->prepareInputForUpdate()
  inc\commondbtm.class.php:1065                      Ticket->prepareInputForUpdate()
  front\ticket.form.php:60                           CommonDBTM->update()
```

  - The document doesn't appears on the web interface
  - The file is correctly stored on C:\xampp\htdocs\glpi\files\DOCX\55\1779a780e69e261fda6616c887e4917d2f4490.DOCX

Regards,
Anael

**Comment by trasher** [comment]
This is not related to #1511 which is a specific master issue (does not happen on 9.1); anyways, those notices are only related to log.

**Comment by gloomy-master** [comment]
Hi all.
What changes are there on this?

**Comment by trasher** [comment]
There are two issues in one here.

The first point has been fixed in the master branch; for the 9.2 release. The second point is not yet fixed I guess.

**Comment by trasher** [comment]
First point is related to #1476

**Comment by trasher** [comment]
I'm not able to reproduce the issue attaching file you provided in a ticket on GLPI 9.1.2

**Comment by trasher** [comment]
Actor issue is fixed for 9.2 (will not be for 9.1.x because it requires a schema change).
Filename issue seems to be solved as of 9.1.2.
